control-plane/api/v1alpha1/api_gateway_types.go

```diff
@@ -65,6 +66,15 @@ type GatewayClassConfigSpec struct {
 
 	// The value to add to privileged ports ( ports < 1024) for gateway containers
 	MapPrivilegedContainerPorts int32 `json:"mapPrivilegedContainerPorts,omitempty"`
+
+	// The value to add pod disruption budgets, will prevent gateway pods from being evicted all at once.
+	PodDisruptionBudgetSpec *PodDisruptionBudgetSpec `json:"podDisruptionBudget,omitempty"`
+}
+
+type PodDisruptionBudgetSpec struct {
+	Enabled        bool                `json:"enabled,omitempty"`
```

Comment:
my thought process was:

1. The configs remain same for resources that we create and the resources created through templates. i.e. they all have - enable, minAvailable and maxUnavailable thus making them homogenous.
2. For the resources we create, if the end-user/dev needs to disable PDB she/he could just go ahead and make enabled = false instead of deleting the whole stanza and adding it back again when they require it.

Shall I go ahead and remove the enabled flag anyways?